# ¿Cuál es la probabilidad de que un hogar mexicano cuente con al menos un automóvil?
Con datos de la Encuesta Nacional de Ingresos y Gastos de los Hogares (ENIGH) 2020

## Metadatos
- **Identifier:** MEX-INEGI.ESD3.03-ENIGH-2020-NS

- **Title:** Encuesta Nacional de Ingresos y Gastos de los Hogares 2020 Nueva Serie. Datos oportunos
- **Description:** INEGI. Encuesta Nacional de Ingresos y Gastos de los Hogares 2020 Nueva Serie. SNIEG. Información de Interés Nacional.

La Encuesta Nacional de Ingresos y Gastos de los Hogares (ENIGH) tiene como objetivo proporcionar un panorama estadístico del comportamiento de los ingresos y gastos de los hogares en cuanto a su monto, procedencia y distribución. Adicionalmente ofrece información sobre las características ocupacionales, sociodemográficas y acceso a alimentación de los integrantes del hogar, así como las características de la infraestructura de la vivienda y el equipamiento del hogar. 

La ENIGH forma parte del sistema de Información de Interés Nacional (IIN), esto significa que los resultados obtenidos de este proyecto son de uso obligatorio para la Federación, los estados y los municipios, a efecto de coadyuvar al desarrollo nacional.
En el año 1984 comenzó una tendencia a ampliar los objetivos y a homogeneizar la metodología, tomando en cuenta las recomendaciones internacionales y los requerimientos de información de los diferentes usuarios, cuidando la comparabilidad histórica. 
- **Periodicidad:** A partir de 1992 se realiza con una periodicidad bienal (cada dos años) con excepción del año 2005 en que se realizó un levantamiento extraordinario. 
- **Población objetivo:** La constituyen los hogares de nacionales o extranjeros, que residen habitualmente en viviendas particulares dentro del territorio nacional. 
- **Unidad de Selección:** Vivienda particular. Las viviendas se eligen mediante un minucioso proceso estadístico que garantiza que los resultados obtenidos a partir de sólo una parte de la población (muestra), se pueda generalizar hacia el total. 
- **Marco Muestral:** El marco de propósitos múltiples del INEGI, está constituido con la información demográfica y cartográfica obtenida en el Censo de Población y Vivienda 2010. 
- **Unidad de observación:** El hogar. 
- **Unidad de análisis:** El hogar, la vivienda y los integrantes del hogar. 
- **Cobertura temática:** 
    - Características de la vivienda. 
    - Residentes e identificación de hogares en la vivienda. 
    - Características sociodemográficas de los residentes de la vivienda. 
    - Equipamiento del hogar, servicios. 
    - Condición de actividad y características ocupacionales de los integrantes del hogar de 12 y más años. 
    - Ingreso corriente total (monetario y no monetario) de los hogares. 
    - Percepciones financieras y de capital de los hogares y sus integrantes. 
    - Gasto corriente monetario de los hogares. 
    - Erogaciones financieras y de capital de los hogares. 
Los diferentes conceptos de la ENIGH están regidos por recomendaciones acordadas en convenciones internacionales por ejemplo: 
Las resoluciones e informes de las 18 Conferencias Internacionales sobre estadísticas del trabajo, de la Organización Internacional del Trabajo (OIT). 

El reporte final y recomendaciones del Grupo de Canberra, grupo de expertos en las "Estadísticas de Ingreso de los Hogares". 
Manual de Encuestas sobre Hogares. Departamento de Asuntos Económicos y Sociales Internacionales, Oficina de Estadística. Naciones Unidas, Nueva York, 1987. 

Así mismo están articuladas con el Sistema de Cuentas Nacionales y con las Encuestas de Hogar que levanta el INEGI. 
Tamaño de la muestra: A nivel nacional son, incluyendo la decena uno, 105 483 viviendas particulares. 
Periodo de levantamiento: La captación de la información se llevará a cabo entre el 21 de agosto al 28 de noviembre de este año. A lo largo de este periodo se efectúan diez cortes, cada uno organizado en diez días; por lo tanto, cada uno de estos cortes los conoceremos como decenas (consulta calendario en el anexo).

- **Carga de trabajo:** De acuerdo a la minuciosidad en el registro de la información en este proyecto se ha definido para cada entrevistador una carga de seis entrevistas en viviendas particulares por decena. El número de entrevistas puede disminuir o aumentar de acuerdo con varios factores: no respuesta, recuperación de la no respuesta u hogares adicionales.  

- **Modified:** 2021-07-28

- **Publisher:** Instituto Nacional de Estadística y Geografía, INEGI

- **Mbox:** atencion.usuarios@inegi.org.mx

- **License:** https://www.inegi.org.mx/inegi/terminos.html

- **DataDictionary:** Keyword: "alimentación", "consumo", "encuesta", "entretenimiento", "gasto", "hogar", "ingreso", "ocupación", "población", "pobreza", "remuneración", "trabajo", "vivienda"

- **ContactPoint:** Mtro. J. Armando Aguiar Rodríguez, 800 111 4634

- **Temporal:** 2020-08-21-2020-11-28

- **Spatial:** Estados Unidos Mexicanos

- **AccrualPeriodicity:** Bienalmente

- **Distribution:** https://www.inegi.org.mx/programas/enigh/nc/2020/

## Variable endógena/dependiente
- *(Dummy)* *__El hogar cuenta con al menos un automóvil.__* 
Se modela a partir del número de autos con los que cuenta un hogar.
    - 1 = El hogar cuenta con al menos un auto
    - 0 = El hogar **NO** cuenta con auto

## Variables exógenas/independientes
1. *(Cuantitativa)* *__Ingresos acumulados del hogar__* Se espera que los ingresos acumulados trimestrales sean un indicador de riqueza y proxy de recursos suficientes para tener un vehículo. **Se contempla una relación positiva entre los ingresos acumulados y la probabilidad de contar con al menos un vehículo.**
2. *(Cuantitativa)* *__Número de integrantes del hogar__* . Económicamente un número mayor de integrantes puede significar menos recursos disponibles para tener un vehículo. Socialmente, el mayor número de integrantes puede significar una mayor necesidad para la seguridad y el transporte de la familia. **Bajo una visión económica se espera una relación inversa entre esta variable y la endógena.**
2. *(Cuantitativa)* *__Numero de fuentes de ingresos del hogar__* Un número mayor de fuentes de ingresos pueden indicar, entre tantas cosas, mayores ingresos percibidos, diversificación de recursos e inversiones, múltiples negocios, etc. **Se espera una relación positiva**
3. *(Dummy)* *__El jefe del hogar es hombre__* El sexo puede ser un determinante para la adquisición de un vehículo, ya sea por una desigualdad salarial, el interés de los sexos por contar con un auto o factores como la cultura o la aversión al conducir. **Se contempla una mayor probabilidad esperada al ser hombre por las distintas desigualdades que se perciben en el país.**
    - 1 = Es hombre
    - 0 = Es mujer
4. *(Cuantitativa)* *__Edad del jef(a) del hogar:__* La edad puede ser un buen determinante pues al ser mayor y tener más experiencia, se esperarían mayores oportunidades como lo son las laborales y salariales. Asimismo, la edad puede significar una acumulación de riqueza durante los años en los que se ha laborado. Estos puntos pueden ofrecer los recursos para contar con un automóvil. **Se espera una relación positiva.** 
5. *(Dummy)* *__El jefe del hogar terminó la secundaria__* El haber terminado la secundaria puede ser un indicador de las oportunidades económicas como lo son las laborales y salariales. La retribución por el trabajo de una persona sin la secundaria terminada puede no ser suficientes para contar con un vehículo. **Se contempla un aumento en la probabilidad de tener un auto al terminar la secundaria.**
    - 1 = El/La jef(a) del hogar terminó la secundaria
    - 0 = El/La jef(a) del hogar **NO** terminó la secundaria
8. *(Cuantitativa)* *__Tasa de ahorro__*. Económicamente, una tasa de ahorro mayor implica un menor consumo, por lo que se podría deducir que no se están consumiendo gran cantidad de productos. Dentro de estos podrían entrar los insumos de un vehículo: gasolina, pago de tenencia, pago de multas, reparaciones, etc. Por lo tanto, **se espera una relación inversa entre la tasa de ahorro y contar con un vehículo.**

# Función esperada:
$tiene\_auto(tasa\_ahorro\_tri, num\_integrantes, es\_hombre, edad\_jefe, termino\_sec, ingtot\_tri, num\_ingresos)$

tiene_auto c es_hombre termino_sec edad_jefe ingtot_tri num_ingresos tasa_ahorro_tri num_integrantes

# Preguntas
- ¿Cuáles son los costos asociados a tener un vehículo?
- ¿La escolaridad es un determinante en la tenencia de un automóvil?
- ¿La tasa de ahorro permite pronosticar la tenencia de automóvil?
- ¿El sexo de los integrantes del hogar es determinante en la tenencia de un vehículo?
- La proporción de gastos en alimentos y transporte público, con respecto al ingreso total del hogar, ¿son determinantes en la tenencia de un vehículo?
- ¿La edad del jefe del hogar es un determinante?
- ¿La cantidad de integrantes del hogar es un determinante?

# Bases de datos

In [1]:
import pandas as pd
import numpy as np

# Bases de datos
hogares = pd.read_csv('bdd/enigh2020/conjunto_de_datos_hogares_enigh_2020_ns.csv')
hogares_gastos = pd.read_csv('bdd/enigh2020/conjunto_de_datos_gastoshogar_enigh_2020_ns.csv')
ingresos = pd.read_csv('bdd/enigh2020/conjunto_de_datos_ingresos_enigh_2020_ns.csv')
poblacion = pd.read_csv('bdd/enigh2020/conjunto_de_datos_poblacion_enigh_2020_ns.csv')
concentrado_hogar = pd.read_csv('bdd/enigh2020/conjunto_de_datos_concentradohogar_enigh_2020_ns.csv')

c:\users\ruser\appdata\local\programs\python\python39\lib\site-packages\IPython\core\interactiveshell.py:3165: DtypeWarning: Columns (13,14,16,20,21,22,23,24,25) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,
c:\users\ruser\appdata\local\programs\python\python39\lib\site-packages\IPython\core\interactiveshell.py:3165: DtypeWarning: Columns (10,11,12,13,14,15,16,17,82,83,166) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


## Puras
Bases de datos sin modificar

In [2]:
hogares

folioviv  foliohog  huespedes huesp_come  num_trab_d trab_come  \
0       100013605         1          0                      0             
1       100013606         1          0                      0             
2       100017801         1          0                      0             
3       100017802         1          0                      0             
4       100017803         1          0                      0             
...           ...       ...        ...        ...         ...       ...   
89001  3260770714         1          0                      0             
89002  3260770715         1          0                      0             
89003  3260770716         1          0                      0             
89004  3260770717         1          0                      0             
89005  3260770718         1          0                      0             

       acc_alim1  acc_alim2  acc_alim3  acc_alim4  ...  est_trans  bene_licon  \
0              1          2          1          1  ...        380           2   
1              1          1          1          1  ...        600           2   
2              2          2          2          2  ...          0           2   
3              2          2          2          2  ...        200           2   
4              2          2          2          2  ...          0           2   
...          ...        ...        ...        ...  ...        ...         ...   
89001          2          2          2          2  ...          0           1   
89002          2          2          2          2  ...          0           1   
89003          1          2          2          1  ...          0           2   
89004          2          2          2          2  ...          0           2   
89005          2          2          2          2  ...          0           1   

      cond_licon lts_licon otros_lts diconsa frec_dicon cond_dicon pago_dicon  \
0                                          9                                    
1                                          9                                    
2                                          9                                    
3                                          9                                    
4                                          2                                    
...          ...       ...       ...     ...        ...        ...        ...   
89001          2         7         2       1          2          2          2   
89002          2         7         2       1          3                         
89003                                      1          3                         
89004                                      1          2          2          3   
89005          2         2                 1          2          2          3   

      otro_pago  
0                
1                
2                
3                
4                
...         ...  
89001            
89002            
89003            
89004            
89005            

[89006 rows x 137 columns]

In [3]:
hogares_gastos

folioviv  foliohog clave tipo_gasto  mes_dia  forma_pag1  \
0         100013605         1  C001         G1        0           1   
1         100013605         1  C004         G1        0           1   
2         100013605         1  C005         G1        0           1   
3         100013605         1  C006         G1        0           1   
4         100013605         1  C015         G1        0           1   
...             ...       ...   ...        ...      ...         ...   
4948439  3260770714         1  G104         G7        0           0   
4948440  3260770715         1  G104         G7        0           0   
4948441  3260770716         1  G105         G7        0           0   
4948442  3260770717         1  G105         G7        0           0   
4948443  3260770718         1  G105         G7        0           0   

         forma_pag2  forma_pag3  lugar_comp  orga_inst  ...  inmujer  inst_1  \
0                 0           0           6          0  ...                    
1                 0           0           6          0  ...                    
2                 0           0           6          0  ...                    
3                 0           0           6          0  ...                    
4                 0           0           6          0  ...                    
...             ...         ...         ...        ...  ...      ...     ...   
4948439           0           0           0          0  ...                    
4948440           0           0           0          0  ...                    
4948441           0           0           0          0  ...                    
4948442           0           0           0          0  ...                    
4948443           0           0           0          0  ...                    

         inst_2 num_meses num_pagos ultim_pago gasto_tri gasto_nm gas_nm_tri  \
0                                                  108.0                       
1                                                   57.0                       
2                                                   69.0                       
3                                                   45.0                       
4                                                   96.0                       
...         ...       ...       ...        ...       ...      ...        ...   
4948439                                                      1000       3000   
4948440                                                      1000       3000   
4948441                                                       500       1500   
4948442                                                      1000       3000   
4948443                                                      1500       4500   

        imujer_tri  
0                   
1                   
2                   
3                   
4                   
...            ...  
4948439             
4948440             
4948441             
4948442             
4948443             

[4948444 rows x 27 columns]

In [4]:
ingresos

folioviv  foliohog  numren clave mes_1 mes_2 mes_3 mes_4 mes_5  \
0        100013605         1       1  P022    10    09    08    07    06   
1        100013605         1       1  P040    10    09    08    07    06   
2        100013605         1       1  P049    10    09    08    07    06   
3        100013605         1       1  P069    10    09    08    07    06   
4        100013605         1       2  P022    10    09    08    07    06   
...            ...       ...     ...   ...   ...   ...   ...   ...   ...   
394907  3260770717         1       4  P001    09    08    07    06    05   
394908  3260770718         1       1  P040    09    08    07    06    05   
394909  3260770718         1       2  P001    09    08    07    06    05   
394910  3260770718         1       3  P001    09    08    07    06    05   
394911  3260770718         1       3  P009                                 

       mes_6  ing_1 ing_2  ing_3 ing_4 ing_5 ing_6   ing_tri  
0         05      0  3200   3200  3200  3200     0   6295.08  
1         05    300   300    300   300   300   300    885.24  
2         05      0     0      0     0     0  1000    491.80  
3         05   3200     0      0     0     0     0   1573.77  
4         05      0     0  11000     0     0     0   5409.83  
...      ...    ...   ...    ...   ...   ...   ...       ...  
394907    04   2800  1700      0     0     0     0   2213.11  
394908    04    500   500    500  1000  1000   500   1967.21  
394909    04   4800  4800   4800  4800  4800  4800  14163.93  
394910    04   7000  7000   7000  7000  7000  7000  20655.73  
394911        10000                                  2459.01  

[394912 rows x 17 columns]

In [5]:
poblacion

folioviv  foliohog  numren  parentesco  sexo  edad  madre_hog  \
0        100013605         1       1         101     2    48          2   
1        100013605         1       2         301     2    17          1   
2        100013605         1       3         301     1    14          1   
3        100013606         1       1         101     1    46          2   
4        100013606         1       2         201     2    43          2   
...            ...       ...     ...         ...   ...   ...        ...   
315738  3260770717         1       6         301     1    12          1   
315739  3260770717         1       7         301     2     5          1   
315740  3260770718         1       1         101     2    64          2   
315741  3260770718         1       2         301     1    33          1   
315742  3260770718         1       3         301     1    31          1   

       madre_id  padre_hog padre_id  ... hijos_viv hijos_mue hijos_sob  \
0                        2           ...         2         0         2   
1            01          2           ...         0                       
2            01          2           ...                                 
3                        2           ...                                 
4                        2           ...         2         0         2   
...         ...        ...      ...  ...       ...       ...       ...   
315738       02          1       01  ...                                 
315739       02          1       01  ...                                 
315740                   2           ...        10         2         8   
315741       01          2           ...                                 
315742       01          2           ...                                 

       trabajo_mp motivo_aus act_pnea1 act_pnea2 num_trabaj c_futuro ct_futuro  
0               1                                         1                     
1               2                    4                                          
2               2                    4                                          
3               1                                         1                     
4               2                    3                                          
...           ...        ...       ...       ...        ...      ...       ...  
315738          2                    4                                          
315739                                                                          
315740          2                    3                                          
315741          1                                         1                     
315742          1                                         1                     

[315743 rows x 184 columns]

In [6]:
concentrado_hogar

folioviv  foliohog  ubica_geo  tam_loc  est_socio  est_dis    upm  \
0       100013605         1       1001        1          3        2      1   
1       100013606         1       1001        1          3        2      1   
2       100017801         1       1001        1          3        2      2   
3       100017802         1       1001        1          3        2      2   
4       100017803         1       1001        1          3        2      2   
...           ...       ...        ...      ...        ...      ...    ...   
89001  3260770714         1      32012        4          2      558  10149   
89002  3260770715         1      32012        4          2      558  10149   
89003  3260770716         1      32012        4          2      558  10149   
89004  3260770717         1      32012        4          2      558  10149   
89005  3260770718         1      32012        4          2      558  10149   

       factor  clase_hog  sexo_jefe  ...  mater_serv  material  servicio  \
0         190          2          2  ...        0.00      0.00       0.0   
1         190          2          1  ...        0.00      0.00       0.0   
2         189          2          1  ...       78.68     78.68       0.0   
3         189          2          1  ...        0.00      0.00       0.0   
4         189          2          1  ...        0.00      0.00       0.0   
...       ...        ...        ...  ...         ...       ...       ...   
89001     128          2          1  ...        0.00      0.00       0.0   
89002     128          2          1  ...        0.00      0.00       0.0   
89003     128          2          1  ...        0.00      0.00       0.0   
89004     128          2          1  ...        0.00      0.00       0.0   
89005     128          2          2  ...        0.00      0.00       0.0   

       deposito  prest_terc  pago_tarje   deudas  balance  otras_erog      smg  
0          0.00        0.00         0.0     0.00     0.00         0.0  11089.8  
1          0.00        0.00         0.0  2360.65     0.00         0.0  11089.8  
2        983.60        0.00         0.0     0.00     0.00         0.0  11089.8  
3          0.00      885.24         0.0     0.00     0.00         0.0  11089.8  
4       5901.63        0.00         0.0     0.00     0.00         0.0  11089.8  
...         ...         ...         ...      ...      ...         ...      ...  
89001      0.00        0.00         0.0     0.00  1106.55         0.0  11089.8  
89002   3442.62        0.00         0.0     0.00  1425.24         0.0  11089.8  
89003   2213.11        0.00         0.0     0.00   879.34         0.0  11089.8  
89004   1377.04        0.00         0.0     0.00     0.00         0.0  11089.8  
89005   4426.22        0.00         0.0  2950.81     0.00         0.0  11089.8  

[89006 rows x 126 columns]

## Manejo de base de datos

### Hogares
#### Variables de interés
1. **folioviv**: Identificador de la vivienda compuesto por dos dígitos con la clave de la entidad federativa, uno con el ámbito (urbano, código diferente a 6; rural, código 6), cuatro dígitos del número consecutivo de la upm, un dígito de la decena de levantamiento y dos dígitos con un número consecutivo para la vivienda seleccionada.
2. **foliohog**: Identificador del hogar, el código 1 identifica al hogar principal y del 2 al 5 los hogares adicionales 
3. **num_auto**: Número de automóviles del hogar
4. **est_alim**: Estimación del gasto mensual en alimentos, bebidas y tabaco consumidos dentro y fuera de casa.
5. **est_trans**: Estimación del gasto mensual en transporte público.

¿La estimaciones son expectativas?
#### Funciones de las variables
- **folioviv** y **foliohog** permiten identificar un mismo hogar y unir los datos con otras bdd
- **num_auto** es la variable eje para el ejercicio, permite conocer si el hogar cuenta con almenos un automóvil y usarla como variable para determinar la probabilidad
- **est_alim y est_trans**: permite dar una idea de la proporción de los gastos en alimentos y transporte con base en el ingreso y determinar si los hogares tienen fondos suficientes y libres para tener un automóvil

In [7]:
# Variables de interes de hogares
hogares_vi = ['folioviv', 'foliohog', 'num_auto', 'est_alim', 'est_trans']
# bdd de hogares manejada
hogares_m = hogares[hogares_vi]
hogares_m

folioviv  foliohog  num_auto  est_alim  est_trans
0       100013605         1         0      1800        380
1       100013606         1         1      6400        600
2       100017801         1         1      3000          0
3       100017802         1         0      4500        200
4       100017803         1         1      8500          0
...           ...       ...       ...       ...        ...
89001  3260770714         1         0      1500          0
89002  3260770715         1         0      2000          0
89003  3260770716         1         0      1500          0
89004  3260770717         1         0      4000          0
89005  3260770718         1         1      3000          0

[89006 rows x 5 columns]

### Gasto de los Hogares
#### Variables de interés
1. **folioviv**: Identificador de la vivienda compuesto por dos dígitos con la clave de la entidad federativa, uno con el ámbito (urbano, código diferente a 6; rural, código 6), cuatro dígitos del número consecutivo de la upm, un dígito de la decena de levantamiento y dos dígitos con un número consecutivo para la vivienda seleccionada.
2. **foliohog**: Identificador del hogar, el código 1 identifica al hogar principal y del 2 al 5 los hogares adicionales 
3. **gasto_tri:** Monto del gasto trimestral por tipo de gasto dentro del hogar

In [8]:
# Variables de interes de gasto de los hogares
hogares_gastos_vi = ['folioviv', 'foliohog', 'gasto_tri']
hogares_gastos_m = hogares_gastos[hogares_gastos_vi]
# Reemplaza los espacios en blanco por Not a Number NaN
hogares_gastos_m = hogares_gastos_m.replace(' ', np.nan)
# Cambia el tipo de valores de gasto_tri a float
hogares_gastos_m = hogares_gastos_m.astype({'gasto_tri':'float'})
hogares_gastos_m

folioviv  foliohog  gasto_tri
0         100013605         1      108.0
1         100013605         1       57.0
2         100013605         1       69.0
3         100013605         1       45.0
4         100013605         1       96.0
...             ...       ...        ...
4948439  3260770714         1        NaN
4948440  3260770715         1        NaN
4948441  3260770716         1        NaN
4948442  3260770717         1        NaN
4948443  3260770718         1        NaN

[4948444 rows x 3 columns]

### Ingresos
#### Variables de interés
1. **folioviv**: Identificador de la vivienda compuesto por dos dígitos con la clave de la entidad federativa, uno con el ámbito (urbano, código diferente a 6; rural, código 6), cuatro dígitos del número consecutivo de la upm, un dígito de la decena de levantamiento y dos dígitos con un número consecutivo para la vivienda seleccionada.
2. **foliohog**: Identificador del hogar, el código 1 identifica al hogar principal y del 2 al 5 los hogares adicionales
3. **numren**: Número consecutivo en el registro de personas del hogar
4. **ing_tri**:  Monto del Ingreso Trimestral por tipo de ingreso en el hogar

#### Funciones de variables
- **folioviv** y **foliohog** permiten identificar un mismo hogar y unir los datos con otras bdd
- **numren** permite identificar a una persona en específica dentro de un mismo hogar
- **ing_tri** permite determinar el monto del ingreso trimestral en el hogar. Aproximador de riqueza.
    - **Nota 1/**: Al observar los registros se observan mútiples registros para una misma persona en un mismo hogar, es decir, hay personas con múltiples tipos de ingresos trimestrales.
    - **Nota 2/**: Al agrupar los hogares por **'folioviv'** y **'foliohog'**, es decir, por hogar único, de **89,006** registros de hogares únicos en bases de datos como **hogares y hogares_gastos**, **ingresos** solo cuenta con registros para **88,926**. En otras palabras, **hay 80 hogares únicos los cuales no cuentan con datos de los ingresos.**

In [9]:
# Variables de interes de ingreso
ingresos_vi = ['folioviv', 'foliohog', 'numren', 'ing_tri']
# bdd de ingresos manejada
ingresos_m = ingresos[ingresos_vi]
ingresos_m

folioviv  foliohog  numren   ing_tri
0        100013605         1       1   6295.08
1        100013605         1       1    885.24
2        100013605         1       1    491.80
3        100013605         1       1   1573.77
4        100013605         1       2   5409.83
...            ...       ...     ...       ...
394907  3260770717         1       4   2213.11
394908  3260770718         1       1   1967.21
394909  3260770718         1       2  14163.93
394910  3260770718         1       3  20655.73
394911  3260770718         1       3   2459.01

[394912 rows x 4 columns]

### Poblacion
#### Variables de interés
1. **folioviv**: Identificador de la vivienda compuesto por dos dígitos con la clave de la entidad federativa, uno con el ámbito (urbano, código diferente a 6; rural, código 6), cuatro dígitos del número consecutivo de la upm, un dígito de la decena de levantamiento y dos dígitos con un número consecutivo para la vivienda seleccionada.
2. **foliohog**: Identificador del hogar, el código 1 identifica al hogar principal y del 2 al 5 los hogares adicionales
3. **numren**: Número consecutivo en el registro de personas del hogar
4. **parentesco**: Categorías en el catálogo de parentesco
    - 101	Jefe(a)
    - 102	Persona sola
    - **Nota 1/**: Se tomarán estas variables para determinar el los detalles del jefe del hogar. Para este trabajo ambas variables serán tomadas por igual. Se considera que una persona que vive sola toma indirectamente el rol de jefe del hogar por no haber otro que tome el papel.
    - **Nota 2/**: Al analizar la base de datos de **poblacion** se descubrió que solo existen registros **(89,006)** con **código 101 (jefes)** mientras que para **102 (persona sola)** no exiten registros.
4. **sexo**: Sexo de la persona
    - 1 = Hombre
    - 2 = Mujer
5. **edad**: Edad de la persona
6. **nivelaprob**: Nivel escolar aprobado
    - 0	Ninguno
    - 1	Preescolar
    - 2	Primaria
    - 3	Secundaria
    - 4	Preparatoria o bachillerato
    - 5	Normal
    - 6	Carrera técnica o comercial
    - 7	Profesional
    - 8	Maestría
    - 9	Doctorado
    
#### Funciones de las variables
- **folioviv** y **foliohog** permiten identificar un mismo hogar y unir los datos con otras bdd
- **numren** permite identificar a una persona en específica dentro de un mismo hogar
- **parentesco** permite identificar, para este trabajo, si se es el jefe del hogar
- **sexo** permite identificar el sexo de una persona
- **edad** permite identificar la edad de una persona
- **nivelaprob** permite identificar, para este trabajo, si los jefes del hogar terminaron la secundaria

In [10]:
# Variables de interes de poblacion
poblacion_vi = ['folioviv', 'foliohog', 'numren', 'parentesco', 'sexo', 'edad', 'nivelaprob']
# Bdd de poblacion manejada
poblacion_m = poblacion[poblacion_vi]
poblacion_m

folioviv  foliohog  numren  parentesco  sexo  edad nivelaprob
0        100013605         1       1         101     2    48          7
1        100013605         1       2         301     2    17          4
2        100013605         1       3         301     1    14          3
3        100013606         1       1         101     1    46          4
4        100013606         1       2         201     2    43          6
...            ...       ...     ...         ...   ...   ...        ...
315738  3260770717         1       6         301     1    12          2
315739  3260770717         1       7         301     2     5          1
315740  3260770718         1       1         101     2    64          3
315741  3260770718         1       2         301     1    33          3
315742  3260770718         1       3         301     1    31          7

[315743 rows x 7 columns]

In [11]:
concentrado_hogar_vi = ['folioviv', 'foliohog', 'tot_integ', 'sexo_jefe', 'edad_jefe', 'educa_jefe', 'alimentos', 'gasto_mon',
                        'transporte', 'ing_cor']
concentrado_hogar_m = concentrado_hogar[concentrado_hogar_vi]
concentrado_hogar_m

folioviv  foliohog  tot_integ  sexo_jefe  edad_jefe  educa_jefe  \
0       100013605         1          3          2         48           9   
1       100013606         1          4          1         46           8   
2       100017801         1          2          1         26          10   
3       100017802         1          2          1         29           8   
4       100017803         1          2          1         63          10   
...           ...       ...        ...        ...        ...         ...   
89001  3260770714         1          2          1         65           4   
89002  3260770715         1          2          1         75           3   
89003  3260770716         1          3          1         52           3   
89004  3260770717         1          7          1         49           4   
89005  3260770718         1          3          2         64           6   

       alimentos  gasto_mon  transporte   ing_cor  
0       14732.80   24626.04     5447.24  16229.49  
1        9321.32   20397.10     4915.68  31425.68  
2       15081.32   44955.73     7029.68  33979.16  
3       26921.53   82950.42     7022.39  71557.37  
4       11969.93   30140.68     7154.75  90703.26  
...          ...        ...         ...       ...  
89001    2712.80    6808.18     1140.00  14227.85  
89002    5952.76    9845.39      150.00  13878.67  
89003    2519.96    5462.66      270.00  20104.84  
89004   12021.26   19219.14     1650.00  25967.20  
89005   14226.24   31743.11     3360.00  45415.16  

[89006 rows x 10 columns]

## Creación de la base de datos del trabajo
Contempla el procesamiento de los datos y el uso de funciones como pivot tables y combinaciones entre bases de datos

In [12]:
# Hogares por variables agregadas (promedio de los registros)
# al ser los mismos valores en distintos registros de un mismo hogar, el promedio de estos da ese mismo valor
hogares_va = pd.pivot_table(hogares, values=['num_auto', 'est_alim', 'est_trans'], index=['folioviv', 'foliohog'],
              aggfunc='mean')

# Hogares Variables argregadas manejadas
hogares_vam = hogares_va.copy()
# Si num_auto es mayor o igual a 1, tiene valor de 1 mientras que si no, tiene valor 0
hogares_vam['num_auto'] = hogares_vam['num_auto'].apply(lambda num_auto: 1 if num_auto >=1 else 0)
# Poner est_alim y est_trans a trimestres, es decir, multiplicarlos por 3 porque estan en meses
hogares_vam[['est_alim','est_trans']] = hogares_vam[['est_alim', 'est_trans']].apply(lambda col: col*3)
# Se cambia el nombre de la columna de est_alim y est_trans a ..._tri de trimestral y de num_auto a tiene_auto
hogares_vam.rename(columns={'est_alim':'est_alim_tri', 'est_trans':'est_trans_tri', 'num_auto':'tiene_auto'}, inplace=True)
hogares_vam

est_alim_tri  est_trans_tri  tiene_auto
folioviv   foliohog                                         
100013605  1                 5400           1140           0
100013606  1                19200           1800           1
100017801  1                 9000              0           1
100017802  1                13500            600           0
100017803  1                25500              0           1
...                           ...            ...         ...
3260770714 1                 4500              0           0
3260770715 1                 6000              0           0
3260770716 1                 4500              0           0
3260770717 1                12000              0           0
3260770718 1                 9000              0           1

[89006 rows x 3 columns]

In [13]:
# Gasto de los hogares por variables sumadas (suma de todos los gastos por hogar)
hogares_gastos_va = pd.pivot_table(hogares_gastos_m, values='gasto_tri', index=['folioviv', 'foliohog'], aggfunc=np.sum)
# cambio de gasto_tri a gastot_tri para indicar el total de gasto acumulado
hogares_gastos_va.columns = ['gastot_tri']
hogares_gastos_va

gastot_tri
folioviv   foliohog            
100013605  1           22693.22
100013606  1           18931.42
100017801  1           44520.13
100017802  1           76500.43
100017803  1           30140.68
...                         ...
3260770714 1            6808.18
3260770715 1            9845.39
3260770716 1            5462.66
3260770717 1           18892.14
3260770718 1           31743.11

[89006 rows x 1 columns]

In [14]:
# Ingresos variables agregadas (conteo de tipos de ingreso y suma de los ingresos)
ingresos_va = pd.pivot_table(ingresos_m, values='ing_tri', index=['folioviv', 'foliohog'],
               aggfunc=[np.sum, 'count'])

# Ingresos variables agregadas manejadas
ingresos_vam = ingresos_va.copy()
# Se cambian las columnas por ingtot_tri: ingreso total trimestral del hogar
# y a num_ingresos: numero de ingresos percibidos en el hogar
ingresos_vam.columns = ['ingtot_tri', 'num_ingresos']
ingresos_vam

ingtot_tri  num_ingresos
folioviv   foliohog                          
100013605  1           16229.49             6
100013606  1           23827.86             3
100017801  1           37421.77             5
100017802  1           71557.37             2
100017803  1           71803.26             3
...                         ...           ...
3260770714 1           11803.26             3
3260770715 1           15304.89             5
3260770716 1            8704.91             2
3260770717 1           22967.20             4
3260770718 1           39245.88             4

[88926 rows x 2 columns]

In [15]:
# Poblacion jefes
poblacion_m
poblacion_jefes = poblacion_m[poblacion_m['parentesco']==101]
poblacion_jefes = poblacion_jefes.astype({'nivelaprob':'int64'})
# Booleano si termino la secundaria, 1 si el nivel es mayor o igual a 3 (secundaria) y 0 si no lo es
poblacion_jefes['termino_sec'] = poblacion_jefes['nivelaprob'].apply(lambda nivel: 1 if nivel >= 3 else 0 )
# Boolean si es hombre, 1 es hombre, 0 es mujer
poblacion_jefes['es_hombre'] = poblacion_jefes['sexo'].apply(lambda sexo: 1 if sexo == 1 else 0)
# Variables de interes de poblacion jefes del hogar
poblacion_jefes_vi = ['folioviv', 'foliohog', 'es_hombre', 'edad', 'termino_sec']
# Poblacion jefes manejada
poblacion_jefes_m = poblacion_jefes[poblacion_jefes_vi].set_index(['folioviv', 'foliohog'])
poblacion_jefes_m.rename(columns={'edad':'edad_jefe'}, inplace=True)
poblacion_jefes_m

es_hombre  edad_jefe  termino_sec
folioviv   foliohog                                   
100013605  1                 0         48            1
100013606  1                 1         46            1
100017801  1                 1         26            1
100017802  1                 1         29            1
100017803  1                 1         63            1
...                        ...        ...          ...
3260770714 1                 1         65            0
3260770715 1                 1         75            0
3260770716 1                 1         52            0
3260770717 1                 1         49            0
3260770718 1                 0         64            1

[89006 rows x 3 columns]

In [16]:
# Poblacion integrantes
# Es la poblacion por el numero de hombres y mujeres contadas
poblacion_integrantes = pd.pivot_table(poblacion_m, values='numren', index=['folioviv', 'foliohog'], columns='sexo', aggfunc='count')
poblacion_integrantes.columns = ['num_hombres', 'num_mujeres']
# Se reemplazan los NaN con 0 para hacer la suma
poblacion_integrantes.replace(np.nan, 0, inplace=True)
# Se crea la columna num_integrantes, es la suma de hombres mas mujeres
poblacion_integrantes['num_integrantes'] = poblacion_integrantes['num_hombres'] + poblacion_integrantes['num_mujeres']
poblacion_integrantes

num_hombres  num_mujeres  num_integrantes
folioviv   foliohog                                           
100013605  1                 1.0          2.0              3.0
100013606  1                 3.0          1.0              4.0
100017801  1                 1.0          1.0              2.0
100017802  1                 2.0          0.0              2.0
100017803  1                 1.0          1.0              2.0
...                          ...          ...              ...
3260770714 1                 1.0          1.0              2.0
3260770715 1                 1.0          1.0              2.0
3260770716 1                 2.0          1.0              3.0
3260770717 1                 3.0          4.0              7.0
3260770718 1                 2.0          1.0              3.0

[89006 rows x 3 columns]

## Base de datos Final

## Base conjunta propia

In [17]:
# Hogares_trabajo
# Une las bases de datos de hogares_vam y hogares_gastos con base en el index (hogar)
hogares_t = pd.merge(hogares_vam, hogares_gastos_va, left_index=True, right_index=True)
# Un la base de datos de hogares_t con los registros de ingresos_vam con base en el hogar único
hogares_t = hogares_t.merge(ingresos_vam, left_index=True, right_index=True)
# Une la base de datos hogares_t con los registros de poblacion_jefes (datos de los jefes del hogar) 
hogares_t = hogares_t.merge(poblacion_jefes_m, left_index=True, right_index=True)
# Une la base de datos hogares_t con los registros de poblacion_integrantes (datos de los integrantes por hogar único)
hogares_t = hogares_t.merge(poblacion_integrantes, left_index=True, right_index=True)
# Crea una tasa de hombres que viven en lugar con respecto al numero de integrantes
hogares_t['tasa_hombres'] = round(hogares_t['num_hombres'] / hogares_t['num_integrantes'], 2)
# Genera la tasa de ahorro de los hogares
hogares_t['tasa_ahorro_tri'] = round((hogares_t['ingtot_tri']-hogares_t['gastot_tri'])/hogares_t['ingtot_tri'], 3)
# Genera la tasa/proporción que se gasta en alimentos estimados con respecto al ingreso, redondeado a 3 decimas
hogares_t['tasa_alim_tri'] = round(hogares_t['est_alim_tri']/hogares_t['ingtot_tri'], 3)
hogares_t.to_csv('bdd/creados/bdd_hogares.csv')
hogares_t

# Comprobacion suma
tiene = hogares_t[hogares_t['tiene_auto']==1].shape[0]
no_tiene = hogares_t[hogares_t['tiene_auto']==0].shape[0]
concuerda = (hogares_t.shape[0] == tiene + no_tiene)

print('Hogares con auto: {:,} | Hogares SIN auto: {:,} | Suma concuerda: {}'.format(tiene, no_tiene, concuerda))

Hogares con auto: 26,880 | Hogares SIN auto: 62,046 | Suma concuerda: True


## Muestra aleatoria para estimación de pendientes Probit y Logit

In [32]:
mualpen = hogares_t.sample(10, random_state=2)[['tiene_auto','termino_sec','gastot_tri','ingtot_tri',
                                               'est_trans_tri','tasa_alim_tri',]]
mualpen.to_csv('bdd/creados/muestra_aleatoria_evpendientes.csv')
mualpen

tiene_auto  termino_sec  gastot_tri  ingtot_tri  \
folioviv   foliohog                                                    
2660021414 1                  0            0    15230.20    11015.20   
2660577904 1                  0            0     6909.67    18635.86   
860048822  1                  1            1    16076.48    18049.16   
2062588003 1                  0            0     7081.65    11759.65   
1560616912 1                  0            1    18708.81    22989.11   
2760272923 1                  1            1    25706.51   100516.29   
2504883315 2                  1            0     9020.70    27586.94   
3061023409 1                  0            0     5118.54     7776.18   
3200210904 1                  1            1    66483.93    98070.64   
505013303  1                  0            1    17271.90    32722.81   

                     est_trans_tri  tasa_alim_tri  
folioviv   foliohog                                
2660021414 1                   690          0.599  
2660577904 1                     0          0.419  
860048822  1                     0          0.249  
2062588003 1                   120          0.306  
1560616912 1                     0          0.352  
2760272923 1                     0          0.060  
2504883315 2                   180          0.087  
3061023409 1                  1500          0.887  
3200210904 1                     0          0.107  
505013303  1                     0          0.425

In [18]:
hogares_t

est_alim_tri  est_trans_tri  tiene_auto  gastot_tri  \
folioviv   foliohog                                                        
100013605  1                 5400           1140           0    22693.22   
100013606  1                19200           1800           1    18931.42   
100017801  1                 9000              0           1    44520.13   
100017802  1                13500            600           0    76500.43   
100017803  1                25500              0           1    30140.68   
...                           ...            ...         ...         ...   
3260770714 1                 4500              0           0     6808.18   
3260770715 1                 6000              0           0     9845.39   
3260770716 1                 4500              0           0     5462.66   
3260770717 1                12000              0           0    18892.14   
3260770718 1                 9000              0           1    31743.11   

                     ingtot_tri  num_ingresos  es_hombre  edad_jefe  \
folioviv   foliohog                                                   
100013605  1           16229.49             6          0         48   
100013606  1           23827.86             3          1         46   
100017801  1           37421.77             5          1         26   
100017802  1           71557.37             2          1         29   
100017803  1           71803.26             3          1         63   
...                         ...           ...        ...        ...   
3260770714 1           11803.26             3          1         65   
3260770715 1           15304.89             5          1         75   
3260770716 1            8704.91             2          1         52   
3260770717 1           22967.20             4          1         49   
3260770718 1           39245.88             4          0         64   

                     termino_sec  num_hombres  num_mujeres  num_integrantes  \
folioviv   foliohog                                                           
100013605  1                   1          1.0          2.0              3.0   
100013606  1                   1          3.0          1.0              4.0   
100017801  1                   1          1.0          1.0              2.0   
100017802  1                   1          2.0          0.0              2.0   
100017803  1                   1          1.0          1.0              2.0   
...                          ...          ...          ...              ...   
3260770714 1                   0          1.0          1.0              2.0   
3260770715 1                   0          1.0          1.0              2.0   
3260770716 1                   0          2.0          1.0              3.0   
3260770717 1                   0          3.0          4.0              7.0   
3260770718 1                   1          2.0          1.0              3.0   

                     tasa_hombres  tasa_ahorro_tri  tasa_alim_tri  
folioviv   foliohog                                                
100013605  1                 0.33           -0.398          0.333  
100013606  1                 0.75            0.205          0.806  
100017801  1                 0.50           -0.190          0.241  
100017802  1                 1.00           -0.069          0.189  
100017803  1                 0.50            0.580          0.355  
...                           ...              ...            ...  
3260770714 1                 0.50            0.423          0.381  
3260770715 1                 0.50            0.357          0.392  
3260770716 1                 0.67            0.372          0.517  
3260770717 1                 0.43            0.177          0.522  
3260770718 1                 0.67            0.191          0.229  

[88926 rows x 15 columns]

### Rebalanceo de 0 a 1

In [19]:
tiene_orig = hogares_t[hogares_t['tiene_auto']==1]
tiene_orig_n = tiene_orig.shape[0]

no_tiene_orig = hogares_t[hogares_t['tiene_auto']==0]
no_tiene_orig_n = no_tiene_orig.shape[0]
concuerda = (hogares_t.shape[0] == tiene_orig_n + no_tiene_orig_n)
print('Hogares con auto: {:,} | Hogares SIN auto: {:,} | Suma concuerda: {}'.format(tiene_orig_n,
                                                                                    no_tiene_orig_n,
                                                                                    concuerda))
# Random state, se utiliza para que al generar una nueva lista aleatoria, se mantenga replicable con este valor
rs = 1
# Toma el dataframe de los registros que no tienen y toma una muestra aleatoria del mismo tamaño que los registros
# que si tienen auto
no_tiene_reb = no_tiene_orig.sample(tiene_orig_n, random_state=rs)
# Base de datos de hogares rebalanceado, combina el original de registros con auto con los registros rebalanceados
# que no tienen
hogares_reb = pd.concat([tiene_orig, no_tiene_reb])
hogares_reb.to_csv('bdd/creados/bdd_hogares_reb_rs{}.csv'.format(rs))
hogares_reb

Hogares con auto: 26,880 | Hogares SIN auto: 62,046 | Suma concuerda: True


est_alim_tri  est_trans_tri  tiene_auto  gastot_tri  \
folioviv   foliohog                                                        
100013606  1                19200           1800           1    18931.42   
100017801  1                 9000              0           1    44520.13   
100017803  1                25500              0           1    30140.68   
100017804  1                13500            480           1    38157.67   
100017805  1                24000              0           1    20512.46   
...                           ...            ...         ...         ...   
1800039904 1                18000           3000           0    46117.79   
960027920  1                14160           1500           0    40251.96   
600181604  1                12000           2400           0    32270.22   
762315816  1                 2400            288           0     4279.17   
1504308901 1                12000              0           0    33055.37   

                     ingtot_tri  num_ingresos  es_hombre  edad_jefe  \
folioviv   foliohog                                                   
100013606  1           23827.86             3          1         46   
100017801  1           37421.77             5          1         26   
100017803  1           71803.26             3          1         63   
100017804  1           30368.84             2          1         33   
100017805  1           40770.48             3          1         60   
...                         ...           ...        ...        ...   
1800039904 1          101494.53             7          0         46   
960027920  1           50429.33             4          1         25   
600181604  1           39688.50             5          0         39   
762315816  1            5114.74             2          1         44   
1504308901 1           47836.93             5          1         63   

                     termino_sec  num_hombres  num_mujeres  num_integrantes  \
folioviv   foliohog                                                           
100013606  1                   1          3.0          1.0              4.0   
100017801  1                   1          1.0          1.0              2.0   
100017803  1                   1          1.0          1.0              2.0   
100017804  1                   1          2.0          2.0              4.0   
100017805  1                   0          2.0          1.0              3.0   
...                          ...          ...          ...              ...   
1800039904 1                   1          2.0          2.0              4.0   
960027920  1                   1          2.0          1.0              3.0   
600181604  1                   1          2.0          2.0              4.0   
762315816  1                   1          1.0          0.0              1.0   
1504308901 1                   1          3.0          4.0              7.0   

                     tasa_hombres  tasa_ahorro_tri  tasa_alim_tri  
folioviv   foliohog                                                
100013606  1                 0.75            0.205          0.806  
100017801  1                 0.50           -0.190          0.241  
100017803  1                 0.50            0.580          0.355  
100017804  1                 0.50           -0.256          0.445  
100017805  1                 0.67            0.497          0.589  
...                           ...              ...            ...  
1800039904 1                 0.50            0.546          0.177  
960027920  1                 0.67            0.202          0.281  
600181604  1                 0.50            0.187          0.302  
762315816  1                 1.00            0.163          0.469  
1504308901 1                 0.43            0.309          0.251  

[53760 rows x 15 columns]

## Rebalanceo de 1 a 0

In [20]:
# Genera el primer data frame con rebalanceo para un cuarto de los registros de los que no tienen
# auto
# Nota: Los registros de no tienen son impares, por lo que se extrajo uno del final para que sean pares
# y los valores de los que si tienen y los que no sean los mismos.
hf_tiene_reb = tiene_orig.sample(int(no_tiene_orig.shape[0]/3), random_state=1)
for i in range(2):
    hf_tiene_reb = pd.concat([hf_tiene_reb, tiene_orig.sample(int(no_tiene_orig.shape[0]/3), random_state=i+1) ])

hf_tiene_reb

est_alim_tri  est_trans_tri  tiene_auto  gastot_tri  \
folioviv   foliohog                                                        
400879401  1                 7500            540           1    16484.42   
805547404  1                21000              0           1    46057.60   
700389203  1                 4500              0           1    28467.64   
401004303  1                 4500              0           1     7545.93   
3202643703 1                 3840           2160           1     8762.10   
...                           ...            ...         ...         ...   
301348413  1                 6000              0           1    46755.59   
1600787003 1                12000              0           1    32433.57   
2260132419 1                 6000              0           1    17442.09   
1302070615 1                 9900              0           1    24988.28   
2503439406 1                12000              0           1    44238.85   

                     ingtot_tri  num_ingresos  es_hombre  edad_jefe  \
folioviv   foliohog                                                   
400879401  1           46709.46            11          1         36   
805547404  1           65837.37            13          1         50   
700389203  1           29423.07             2          1         68   
401004303  1            8804.34             1          1         30   
3202643703 1           11980.32             2          1         49   
...                         ...           ...        ...        ...   
301348413  1           26902.16             2          1         42   
1600787003 1           66228.24             9          0         36   
2260132419 1           21277.17             4          1         35   
1302070615 1           26016.38             3          1         34   
2503439406 1           73027.13             7          1         32   

                     termino_sec  num_hombres  num_mujeres  num_integrantes  \
folioviv   foliohog                                                           
400879401  1                   1          2.0          2.0              4.0   
805547404  1                   1          3.0          1.0              4.0   
700389203  1                   1          2.0          1.0              3.0   
401004303  1                   1          1.0          3.0              4.0   
3202643703 1                   0          1.0          1.0              2.0   
...                          ...          ...          ...              ...   
301348413  1                   1          3.0          1.0              4.0   
1600787003 1                   1          3.0          1.0              4.0   
2260132419 1                   1          1.0          0.0              1.0   
1302070615 1                   1          1.0          4.0              5.0   
2503439406 1                   1          1.0          2.0              3.0   

                     tasa_hombres  tasa_ahorro_tri  tasa_alim_tri  
folioviv   foliohog                                                
400879401  1                 0.50            0.647          0.161  
805547404  1                 0.75            0.300          0.319  
700389203  1                 0.67            0.032          0.153  
401004303  1                 0.25            0.143          0.511  
3202643703 1                 0.50            0.269          0.321  
...                           ...              ...            ...  
301348413  1                 0.75           -0.738          0.223  
1600787003 1                 0.75            0.510          0.181  
2260132419 1                 1.00            0.180          0.282  
1302070615 1                 0.20            0.040          0.381  
2503439406 1                 0.33            0.394          0.164  

[62046 rows x 15 columns]

In [21]:
# Se juntan los registros que tienen automovil rebalanceado y los registros que no tienen originales
hf_reb2 = pd.concat([hf_tiene_reb, no_tiene_orig])
hf_reb2.to_csv('bdd/creados/bdd_hogares_reb2.csv')
hf_reb2

est_alim_tri  est_trans_tri  tiene_auto  gastot_tri  \
folioviv   foliohog                                                        
400879401  1                 7500            540           1    16484.42   
805547404  1                21000              0           1    46057.60   
700389203  1                 4500              0           1    28467.64   
401004303  1                 4500              0           1     7545.93   
3202643703 1                 3840           2160           1     8762.10   
...                           ...            ...         ...         ...   
3260770713 1                 7500              0           0     3063.16   
3260770714 1                 4500              0           0     6808.18   
3260770715 1                 6000              0           0     9845.39   
3260770716 1                 4500              0           0     5462.66   
3260770717 1                12000              0           0    18892.14   

                     ingtot_tri  num_ingresos  es_hombre  edad_jefe  \
folioviv   foliohog                                                   
400879401  1           46709.46            11          1         36   
805547404  1           65837.37            13          1         50   
700389203  1           29423.07             2          1         68   
401004303  1            8804.34             1          1         30   
3202643703 1           11980.32             2          1         49   
...                         ...           ...        ...        ...   
3260770713 1           10032.76             4          1         88   
3260770714 1           11803.26             3          1         65   
3260770715 1           15304.89             5          1         75   
3260770716 1            8704.91             2          1         52   
3260770717 1           22967.20             4          1         49   

                     termino_sec  num_hombres  num_mujeres  num_integrantes  \
folioviv   foliohog                                                           
400879401  1                   1          2.0          2.0              4.0   
805547404  1                   1          3.0          1.0              4.0   
700389203  1                   1          2.0          1.0              3.0   
401004303  1                   1          1.0          3.0              4.0   
3202643703 1                   0          1.0          1.0              2.0   
...                          ...          ...          ...              ...   
3260770713 1                   0          1.0          1.0              2.0   
3260770714 1                   0          1.0          1.0              2.0   
3260770715 1                   0          1.0          1.0              2.0   
3260770716 1                   0          2.0          1.0              3.0   
3260770717 1                   0          3.0          4.0              7.0   

                     tasa_hombres  tasa_ahorro_tri  tasa_alim_tri  
folioviv   foliohog                                                
400879401  1                 0.50            0.647          0.161  
805547404  1                 0.75            0.300          0.319  
700389203  1                 0.67            0.032          0.153  
401004303  1                 0.25            0.143          0.511  
3202643703 1                 0.50            0.269          0.321  
...                           ...              ...            ...  
3260770713 1                 0.50            0.695          0.748  
3260770714 1                 0.50            0.423          0.381  
3260770715 1                 0.50            0.357          0.392  
3260770716 1                 0.67            0.372          0.517  
3260770717 1                 0.43            0.177          0.522  

[124092 rows x 15 columns]

## Base con concentrado hogares

In [22]:
concentrado_hogar_vam = concentrado_hogar_m.copy()
concentrado_hogar_vam['jefe_termino_sec'] = concentrado_hogar_m['educa_jefe'].apply(lambda niv: 1 if niv >= 3 else 0)
concentrado_hogar_vam['sexo_jefe'].replace({2:0}, inplace=True)
tasa_ahorro = round((concentrado_hogar_vam['ing_cor'] - concentrado_hogar_vam['gasto_mon'])/concentrado_hogar_vam['ing_cor'], 3)
concentrado_hogar_vam['tasa_ahorro'] = tasa_ahorro
concentrado_hogar_vam.columns = concentrado_hogar_vam.rename(columns={'sexo_jefe':'jefe_es_hombre'}).columns
hogares_f2 = pd.merge(concentrado_hogar_vam, hogares_vam.reset_index(), on=['folioviv', 'foliohog'])
hogares_f2 = hogares_f2[['folioviv', 'foliohog', 'tot_integ','edad_jefe','jefe_es_hombre', 'alimentos', 'gasto_mon', 'transporte', 'ing_cor', 'jefe_termino_sec', 'tiene_auto',
                        'tasa_ahorro']]
hogares_f2.to_csv('bdd/creados/hogares_f2.csv')
hogares_f2

folioviv  foliohog  tot_integ  edad_jefe  jefe_es_hombre  alimentos  \
0       100013605         1          3         48               0   14732.80   
1       100013606         1          4         46               1    9321.32   
2       100017801         1          2         26               1   15081.32   
3       100017802         1          2         29               1   26921.53   
4       100017803         1          2         63               1   11969.93   
...           ...       ...        ...        ...             ...        ...   
89001  3260770714         1          2         65               1    2712.80   
89002  3260770715         1          2         75               1    5952.76   
89003  3260770716         1          3         52               1    2519.96   
89004  3260770717         1          7         49               1   12021.26   
89005  3260770718         1          3         64               0   14226.24   

       gasto_mon  transporte   ing_cor  jefe_termino_sec  tiene_auto  \
0       24626.04     5447.24  16229.49                 1           0   
1       20397.10     4915.68  31425.68                 1           1   
2       44955.73     7029.68  33979.16                 1           1   
3       82950.42     7022.39  71557.37                 1           0   
4       30140.68     7154.75  90703.26                 1           1   
...          ...         ...       ...               ...         ...   
89001    6808.18     1140.00  14227.85                 1           0   
89002    9845.39      150.00  13878.67                 1           0   
89003    5462.66      270.00  20104.84                 1           0   
89004   19219.14     1650.00  25967.20                 1           0   
89005   31743.11     3360.00  45415.16                 1           1   

       tasa_ahorro  
0           -0.517  
1            0.351  
2           -0.323  
3           -0.159  
4            0.668  
...            ...  
89001        0.521  
89002        0.291  
89003        0.728  
89004        0.260  
89005        0.301  

[89006 rows x 12 columns]

## Rebalanceo de 0 a 1

In [23]:
hf2_tiene_orig = hogares_f2[hogares_f2['tiene_auto']==1]
hf2_ntiene_orig = hogares_f2[hogares_f2['tiene_auto']==0]

hf2_ntiene_reb = hf2_ntiene_orig.sample(hf2_tiene_orig.shape[0], random_state=1)
hf2_reb = pd.concat([hf2_tiene_orig, hf2_ntiene_reb])
hf2_reb.to_csv('bdd/creados/hogares_f2_reb.csv')
hf2_reb

folioviv  foliohog  tot_integ  edad_jefe  jefe_es_hombre  alimentos  \
1       100013606         1          4         46               1    9321.32   
2       100017801         1          2         26               1   15081.32   
4       100017803         1          2         63               1   11969.93   
5       100017804         1          4         33               1    7547.03   
6       100017805         1          3         60               1   11281.98   
...           ...       ...        ...        ...             ...        ...   
81356  3005154905         1          2         31               0    9584.83   
67202  2403943910         1          9         51               1    8087.07   
17336   601438313         1          2         31               1    5766.22   
79923  2901558904         1          3         47               0    6647.04   
49688  1800034704         1          4         56               1   24987.53   

       gasto_mon  transporte    ing_cor  jefe_termino_sec  tiene_auto  \
1       20397.10     4915.68   31425.68                 1           1   
2       44955.73     7029.68   33979.16                 1           1   
4       30140.68     7154.75   90703.26                 1           1   
5       39991.94    16171.31   30368.84                 1           1   
6       20512.46     4200.00   52770.48                 1           1   
...          ...         ...        ...               ...         ...   
81356   20326.57     3787.68   38618.27                 1           0   
67202   16630.04      290.32   35609.36                 1           0   
17336   10340.99      837.78   27906.15                 1           0   
79923   32866.75     1917.08   27793.95                 1           0   
49688   42684.65     6269.99  169795.03                 1           0   

       tasa_ahorro  
1            0.351  
2           -0.323  
4            0.668  
5           -0.317  
6            0.611  
...            ...  
81356        0.474  
67202        0.533  
17336        0.629  
79923       -0.183  
49688        0.749  

[53778 rows x 12 columns]

In [24]:
comp = pd.merge(hogares_f2, hogares_t.reset_index(), on=['folioviv', 'foliohog'])[
    ['folioviv', 'foliohog', 'tiene_auto_x', 'tiene_auto_y', 'termino_sec', 'jefe_termino_sec']]
comp

folioviv  foliohog  tiene_auto_x  tiene_auto_y  termino_sec  \
0       100013605         1             0             0            1   
1       100013606         1             1             1            1   
2       100017801         1             1             1            1   
3       100017802         1             0             0            1   
4       100017803         1             1             1            1   
...           ...       ...           ...           ...          ...   
88921  3260770714         1             0             0            0   
88922  3260770715         1             0             0            0   
88923  3260770716         1             0             0            0   
88924  3260770717         1             0             0            0   
88925  3260770718         1             1             1            1   

       jefe_termino_sec  
0                     1  
1                     1  
2                     1  
3                     1  
4                     1  
...                 ...  
88921                 1  
88922                 1  
88923                 1  
88924                 1  
88925                 1  

[88926 rows x 6 columns]

## Rebalanceo de 1 a 0

In [25]:
# Genera el primer data frame con rebalanceo para un cuarto de los registros de los que no tienen
# auto
# Nota: Los registros de no tienen son impares, por lo que se extrajo uno del final para que sean pares
# y los valores de los que si tienen y los que no sean los mismos.
hf2_tiene_reb = hf2_tiene_orig.sample(int(hf2_ntiene_orig.iloc[:-1].shape[0]/4), random_state=1)
# Genera los siguientes data frame 1/4 mediante un loop
for i in range(2, 5):
    hf2_tiene_reb = pd.concat([hf2_tiene_reb, hf2_tiene_orig.sample(int(hf2_ntiene_orig.iloc[:-1].shape[0]/4), random_state=i)])
hf2_tiene_reb

folioviv  foliohog  tot_integ  edad_jefe  jefe_es_hombre  alimentos  \
43743  1560271918         1          3         45               1   11417.08   
38622  1361477507         1          4         53               1    8003.41   
63931  2300351201         1          1         24               1    9141.36   
17007   601279910         1          4         89               1   10105.53   
26525   910464003         1          5         66               0   11121.32   
...           ...       ...        ...        ...             ...        ...   
28490  1000646506         1          4         45               1    4191.38   
4815    260021720         1          3         42               1   11892.75   
65574  2360030012         1          4         44               1   10697.04   
84348  3102242201         1          3         41               1    8785.16   
83538  3063536510         1          5         61               1   19221.24   

       gasto_mon  transporte    ing_cor  jefe_termino_sec  tiene_auto  \
43743   53206.28    14957.62   75598.23                 1           1   
38622   17757.20     4835.93   29045.31                 1           1   
63931   53083.90     7624.51   77047.13                 1           1   
17007   19175.25     4103.22   35301.97                 1           1   
26525   21379.37     1216.00   27492.27                 0           1   
...          ...         ...        ...               ...         ...   
28490   16159.30     4319.04   61752.45                 1           1   
4815    49848.14    25013.11   38631.14                 1           1   
65574   34793.25    21078.88   51778.05                 1           1   
84348   40655.50    19186.02   85367.86                 1           1   
83538  125642.28    22818.11  143157.39                 1           1   

       tasa_ahorro  
43743        0.296  
38622        0.389  
63931        0.311  
17007        0.457  
26525        0.222  
...            ...  
28490        0.738  
4815        -0.290  
65574        0.328  
84348        0.524  
83538        0.122  

[62116 rows x 12 columns]

In [26]:
# Une los registros de no tiene auto original y el rebalanceo de tiene
hf2_reb2 = pd.concat([hf2_ntiene_orig.iloc[:-1], hf2_tiene_reb])
hf2_reb2.to_csv('bdd/creados/hf2_reb2.csv')
hf2_reb2

folioviv  foliohog  tot_integ  edad_jefe  jefe_es_hombre  alimentos  \
0       100013605         1          3         48               0   14732.80   
3       100017802         1          2         29               1   26921.53   
10      100021404         1          6         76               1   10671.16   
11      100021405         1          1         79               0    1266.40   
12      100021406         1          1         37               1    8254.17   
...           ...       ...        ...        ...             ...        ...   
28490  1000646506         1          4         45               1    4191.38   
4815    260021720         1          3         42               1   11892.75   
65574  2360030012         1          4         44               1   10697.04   
84348  3102242201         1          3         41               1    8785.16   
83538  3063536510         1          5         61               1   19221.24   

       gasto_mon  transporte    ing_cor  jefe_termino_sec  tiene_auto  \
0       24626.04     5447.24   16229.49                 1           0   
3       82950.42     7022.39   71557.37                 1           0   
10      20386.01      925.71   52621.55                 1           0   
11       2208.20      576.00   24001.00                 1           0   
12      32516.62     2849.98   18592.52                 1           0   
...          ...         ...        ...               ...         ...   
28490   16159.30     4319.04   61752.45                 1           1   
4815    49848.14    25013.11   38631.14                 1           1   
65574   34793.25    21078.88   51778.05                 1           1   
84348   40655.50    19186.02   85367.86                 1           1   
83538  125642.28    22818.11  143157.39                 1           1   

       tasa_ahorro  
0           -0.517  
3           -0.159  
10           0.613  
11           0.908  
12          -0.749  
...            ...  
28490        0.738  
4815        -0.290  
65574        0.328  
84348        0.524  
83538        0.122  

[124232 rows x 12 columns]